lint/linter.go

```diff
@@ -63,16 +63,48 @@ var (
 func (l *Linter) Lint(packages [][]string, ruleSet []Rule, config Config) (<-chan Failure, error) {
 	failures := make(chan Failure)
 
+	perModVersions := make(map[string]*goversion.Version)
+	perPkgVersions := make([]*goversion.Version, len(packages))
+	for n, files := range packages {
```

Comment:
:thinking: why do you try to detect the Go version per package?
Is it related to sub-modules?

Reply:
Revive's CLI makes it possible to specify many unrelated directories to lint. In the extreme case, every package may come from a different module. Thus per-package detection.